tests/Config/ConfigResolverTest.php

```diff
@@ -0,0 +1,299 @@
+    {
+        $configFilePath = sprintf(
+            '%s/foo',
+            self::temporaryDirectory(),
+        );
+
+        self::fileSystem()->dumpFile(
+            $configFilePath,
+            ''
+        );
+
+        $configResolver = new Twigcs\Config\ConfigResolver(
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            [
+                'config' => $configFilePath,
+            ]
+        );
+
+        $this->expectException(Twigcs\Config\InvalidConfigurationException::class);
+
+        $configResolver->getConfigFile();
+    }
+
+    public function testGetConfigFileReturnsConfigFileWhenConfigOptionHasBeenSpecifiedAndConfigFileExistsAndReturnsConfig(): void
+    {
+        $configFilePath = sprintf(
+            '%s/foo',
+            self::temporaryDirectory(),
+        );
+
+        $configFileContent = <<<PHP
+<?php
+
+return FriendsOfTwig\Twigcs\Config\Config::create();
+PHP;
+
+        self::fileSystem()->dumpFile(
+            $configFilePath,
+            $configFileContent
+        );
+
+        $configResolver = new Twigcs\Config\ConfigResolver(
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            [
+                'config' => $configFilePath,
+            ]
+        );
+
+        self::assertSame($configFilePath, $configResolver->getConfigFile());
+    }
+
+    /**
+     * @dataProvider provideDefaultConfigFileName
+     */
+    public function testGetConfigFileReturnsConfigFileWhenConfigOptionHasBeenSpecifiedAndBothConfigFileAndDefaultConfigFileExistAndReturnConfigs(string $defaultConfigFileName): void
+    {
+        $configFilePath = sprintf(
+            '%s/foo',
+            self::temporaryDirectory(),
+        );
+
+        $defaultConfigFilePath = sprintf(
+            '%s/%s',
+            self::temporaryDirectory(),
+            $defaultConfigFileName,
+        );
+
+        $configFileContent = <<<PHP
+<?php
+
+return FriendsOfTwig\Twigcs\Config\Config::create();
+PHP;
+
+        self::fileSystem()->dumpFile(
+            $configFilePath,
+            $configFileContent
+        );
+
+        self::fileSystem()->dumpFile(
+            $defaultConfigFilePath,
+            $configFileContent
+        );
+
+        $configResolver = new Twigcs\Config\ConfigResolver(
+            $this->createStub(Twigcs\Container::class),
+            self::temporaryDirectory(),
+            [
+                'config' => $configFilePath,
+            ]
+        );
+
+        self::assertSame($configFilePath, $configResolver->getConfigFile());
+    }
+
+    /**
+     * @return \Generator<string, array{0: string}>
+     */
+    public function provideDefaultConfigFileName(): \Generator
+    {
+        $defaultConfigFileNames = [
+            '.twig_cs',
+            '.twig_cs.dist',
```

Comment:
@Ciloe @OwlyCode @yguedidi

In the context of #247, we could add more data sets here to ensure that these default configuration files are resolved - what do you think?